AlphaWallet/TokenScriptClient/Models/TokenScriptSignatureVerifier.swift

```diff
@@ -2,16 +2,60 @@
 
 import Foundation
 import PromiseKit
+import Alamofire
 
 class TokenScriptSignatureVerifier {
-    //TODO implement signature verification
+
     func verify(xml: String, isOfficial: Bool) -> Promise<TokenScriptSignatureVerificationType> {
         return Promise { seal in
             if isOfficial {
-                seal.fulfill(.verified(domainName: "alphawallet.com"))
+                verifyXMLSignatureViaAPI(xml: xml) { passed in
+                    if passed {
+                        //TODO get domain from validator
+                        seal.fulfill(.verified(domainName: "alphawallet.com"))
```

Comment:
Can you replace the text "alphawallet.com" with the Common Name extracted from the verification API?